Description: Don't allow building bundled libraries
 We dropped the bundled libraries from the source package, we can't build them.
 Drop any reference to them.
Author: Ryan Kavanagh <rak@debian.org>
Origin: vendor
Forwarded: not-needed
Last-Update: 2012-07-31
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: bzflag/configure.ac
===================================================================
--- bzflag.orig/configure.ac	2012-07-31 13:44:22.076787684 -0400
+++ bzflag/configure.ac	2012-07-31 13:46:39.425468763 -0400
@@ -160,34 +160,37 @@
 # do we want the bzflag  client?
 AC_ARG_ENABLE(client, [  --disable-client        server-only build])
 
-# compile our own ares
-AC_ARG_ENABLE(ares_build, [  --enable-ares-build     Force compilation of ares])
-
-# compile our own curl
-AC_ARG_ENABLE(curl_build, [  --enable-curl-build     Force compilation of curl])
-
-# compile our own glew
-AC_ARG_WITH(included-glew,
-        AC_HELP_STRING([--with-included-glew],[build with included GLEW library]),
-        [enable_glew_build="$withval"], )
-
-# compile our own regex
-AC_ARG_ENABLE(regex_build, [  --enable-regex-build    Force compilation of regex])
-
-# compile our own zlib
-AC_ARG_ENABLE(zlib_build, [  --enable-zlib-build     Force compilation of zlib])
-
-# compile everything bz provides
-AC_ARG_ENABLE(all_builds, [  --enable-all-builds     Force compilation of all dependencies])
-
-# enable compilation of all bundled external dependencies
-if test "x$enable_all_builds" = "xyes" ; then
-    enable_ares_build=yes
-    enable_curl_build=yes
-    enable_glew_build=yes
-    enable_regex_build=yes
-    enable_zlib_build=yes
-fi
+## Don't build any bundled libraries for Debian, we dropped them in a repack and
+## should use our own.
+##
+## # compile our own ares
+## AC_ARG_ENABLE(ares_build, [  --enable-ares-build     Force compilation of ares])
+##
+## # compile our own curl
+## AC_ARG_ENABLE(curl_build, [  --enable-curl-build     Force compilation of curl])
+##
+## # compile our own glew
+## AC_ARG_WITH(included-glew,
+##         AC_HELP_STRING([--with-included-glew],[build with included GLEW library]),
+##         [enable_glew_build="$withval"], )
+##
+## # compile our own regex
+## AC_ARG_ENABLE(regex_build, [  --enable-regex-build    Force compilation of regex])
+##
+## # compile our own zlib
+## AC_ARG_ENABLE(zlib_build, [  --enable-zlib-build     Force compilation of zlib])
+##
+## # compile everything bz provides
+## AC_ARG_ENABLE(all_builds, [  --enable-all-builds     Force compilation of all dependencies])
+##
+## # enable compilation of all bundled external dependencies
+## if test "x$enable_all_builds" = "xyes" ; then
+##     enable_ares_build=yes
+##     enable_curl_build=yes
+##     enable_glew_build=yes
+##     enable_regex_build=yes
+##     enable_zlib_build=yes
+## fi
 
 # enable UPnP for server
 AC_ARG_ENABLE(UPnP,       [  --enable-UPnP           Use UPnP for server],
@@ -944,6 +947,7 @@
 AC_MSG_CHECKING(whether to build the included ares lookup library)
 BUILD_ARES="no (using system)"
 if test "x$ares_works" = "xno" ; then
+    AC_MSG_ERROR([Could not find libcares, aborting.])
     AC_DEFINE(BUILD_ARES, 1, [Build the included ares lookup library])
     BUILD_ARES="yes"
     LIBCARES='${top_builddir}/src/other/ares/libcares.la'
@@ -988,6 +992,7 @@
 AC_MSG_CHECKING(whether to build the included regular expression library)
 BUILD_REGEX="no (using system)"
 if test "x$regex_works" = "xno" ; then
+    AC_MSG_ERROR([Could not find regular expression library, aborting.])
     AC_DEFINE(BUILD_REGEX, 1, [Build the included regular expression library])
     BUILD_REGEX="yes"
     LIBREGEX='${top_builddir}/src/other/regex/libregex.la'
@@ -1002,6 +1007,7 @@
 AC_MSG_CHECKING(whether to build the included zlib compression library)
 BUILD_ZLIB="no (using system)"
 if test "x$LIBZ" = "x" ; then
+    AC_MSG_ERROR([Could not find zlib, aborting.])
     AC_DEFINE(BUILD_ZLIB, 1, [Build the included zlib compression library])
     BUILD_ZLIB="yes"
     LIBZ='${top_builddir}/src/other/zlib/libz.la'
@@ -1022,6 +1028,7 @@
 fi
 AC_MSG_CHECKING(whether to build the included curl library)
 if test "x$LIBCURL" = "x" ; then
+    AC_MSG_ERROR([Could not find curl library, aborting.])
     AC_DEFINE(BUILD_CURL, 1, [Build the included curl library])
     BUILD_CURL="yes"
     LIBCURL='${top_builddir}/src/other/curl/lib/libcurl.la'
@@ -1059,6 +1066,7 @@
     if test "x$enable_glew_build" = xno ; then
 	BUILD_GLEW=no
     else
+        AC_MSG_ERROR([Could not find glew library, aborting.])
 	BUILD_GLEW="yes"
 	CONF_CPPFLAGS="$CONF_CPPFLAGS -I\$(top_srcdir)/src/other/glew/include"
     fi
@@ -1099,21 +1107,21 @@
 # for subconfigure
 bz_configure_args="$ac_configure_args"
 
-AC_CONFIG_SUBDIRS(src/other/ares)
-if test "x$BUILD_ARES$BUILD_CURL" = "xyesyes" ; then
-    # instruct curl to use the included ares
-    ac_configure_args="$ac_configure_args --enable-ares --disable-manual"
-    case $host_os in
-	solaris*)
-	    if test "$GCC" != yes; then
-	        # Solaris 10 SSL libraries are compatible only with gcc
-	        ac_configure_args="$ac_configure_args --without-ssl"
-	    fi
-	    ;;
-    esac
-fi
-
-AC_CONFIG_SUBDIRS(src/other/curl)
+##AC_CONFIG_SUBDIRS(src/other/ares)
+##if test "x$BUILD_ARES$BUILD_CURL" = "xyesyes" ; then
+##    # instruct curl to use the included ares
+##    ac_configure_args="$ac_configure_args --enable-ares --disable-manual"
+##    case $host_os in
+##	solaris*)
+##	    if test "$GCC" != yes; then
+##	        # Solaris 10 SSL libraries are compatible only with gcc
+##	        ac_configure_args="$ac_configure_args --without-ssl"
+##	    fi
+##	    ;;
+##    esac
+##fi
+##
+##AC_CONFIG_SUBDIRS(src/other/curl)
 
 AC_CONFIG_FILES([
 	BZFlag.xcodeproj/Makefile
@@ -1182,9 +1190,6 @@
 	src/obstacle/Makefile
 	src/ogl/Makefile
 	src/platform/MacOSX/BZFlag-Info.plist
-	src/other/Makefile
-	src/other/regex/Makefile
-	src/other/zlib/Makefile
 	src/platform/MacOSX/Makefile
 	src/platform/Makefile
 	src/scene/Makefile
Index: bzflag/src/Makefile.am
===================================================================
--- bzflag.orig/src/Makefile.am	2012-07-31 13:44:22.076787684 -0400
+++ bzflag/src/Makefile.am	2012-07-31 13:44:32.652840126 -0400
@@ -18,7 +18,6 @@
 
 # "other" must be near the top, if not first, for maintainer-clean to work
 SUBDIRS = \
-	other \
 	3D \
 	common \
 	net \
